### DISCUSSION TITLE
Can the various components of this library be dynamically updated?

### DISCUSSION THREAD

**Discussion by coquizen** [pterm/pterm#241] [Q&A] [answered]
For example, say I have a table with an arbitrary number of rows. As new data comes in (ie from an API), insert the incoming data as a new row. Is this possible?

**Comment by MarvinJWendt** [comment, marked as answer]
Hi @CaninoDev, yes this is possible!
You can use the `AreaPrinter` for that: https://pterm.sh/#/docs/printer/area

The area printer has an `Update` method, where you can give the new content as parameter, which will then override the previous text. You can see an [example](https://github.com/pterm/pterm/blob/master/_examples/area/main.go) in the link I posted above, which can easily be adapted to work with a `TablePrinter`. Let me know if you need some help :)
